src/score.rs

```diff
@@ -0,0 +1,198 @@
+use crate::{
+    error::MqaError,
+    helpers::parse_graphs,
+    quality_measurements::{
+        datasets, distributions, quality_measurements, QualityMeasurementValue,
+    },
+};
+use oxigraph::{
+    model::{NamedNode, NamedOrBlankNode, NamedOrBlankNodeRef},
+    store::Store,
+};
+use std::collections::HashMap;
+
+pub type DistributionScores = Vec<(NamedOrBlankNode, DimensionScores)>;
+pub type DimensionScores = Vec<(NamedNode, MetricScores)>;
+pub type MetricScores = Vec<(NamedNode, Option<u64>)>;
+
+/// Parses graph and calculates score for all metrics in all dimensions, for all distributions.
+pub fn parse_graph_and_calculate_score(
+    graph: String,
+    scores: &crate::score_graph::DimensionScores,
+) -> Result<DistributionScores, MqaError> {
+    parse_graphs(vec![graph]).and_then(|store| calculate_score(&store, scores))
+}
+
+/// Calculates score for all metrics in all dimensions, for all distributions.
+fn calculate_score(
+    store: &Store,
+    scores: &crate::score_graph::DimensionScores,
+) -> Result<DistributionScores, MqaError> {
+    let graph_measurements = quality_measurements(store)?;
+
+    let dataset = match datasets(store)?.into_iter().next() {
+        Some(dataset) => Ok(dataset),
+        None => Err(MqaError::from("store has no dataset")),
+    }?;
+    let dataset_score = node_score(scores, &graph_measurements, dataset.as_ref());
+
+    let dists = distributions(store)?;
+    Ok(dists
```

Comment:
Den har skal vel returnere en DimensionScores

Reply:
ATM regner den ut score for alle distribusjoner.